tests/test_georaster.py

```diff
@@ -942,20 +942,128 @@ def test_reproject(self, example: str) -> None:
+        assert (intersection_match[2] - r.bounds[0]) % r.res[0] == 0
+        assert (intersection_match[1] - r.bounds[3]) % r.res[0] == 0
+        assert (intersection_match[3] - r.bounds[3]) % r.res[1] == 0
+
+        # -- Test with a second raster shifted right and up, integer shift -- #
+        transform_shifted = (
+            r.transform.a,
+            r.transform.b,
+            r.transform.c + rand_int * r.res[0],
+            r.transform.d,
+            r.transform.e,
+            r.transform.f + rand_int * abs(r.res[1]),
+        )
+        r_shifted = gu.Raster.from_array(r.data, crs=r.crs, transform=transform_shifted, nodata=r.nodata)
+        intersection = r.intersection(r_shifted)
+
+        # left and bottom bounds should correspond to shifted raster
+        assert intersection[0] == r_shifted.bounds[0]
+        assert intersection[1] == r_shifted.bounds[1]
+
+        # right and top bounds should correspond to original raster
+        assert intersection[2] == r.bounds[2]
+        assert intersection[3] == r.bounds[3]
+
+        # -- Test with a second raster shifted right and up, float shift -- #
+        transform_shifted = (
+            r.transform.a,
+            r.transform.b,
+            r.transform.c + rand_float * r.res[0],
+            r.transform.d,
+            r.transform.e,
+            r.transform.f + rand_float * abs(r.res[1]),
+        )
+        r_shifted = gu.Raster.from_array(r.data, crs=r.crs, transform=transform_shifted, nodata=r.nodata)
+
+        # With match_ref = False, same as with integer
+        intersection = r.intersection(r_shifted, match_ref=False)
+        assert intersection[0] == r_shifted.bounds[0]
+        assert intersection[1] == r_shifted.bounds[1]
+        assert intersection[2] == r.bounds[2]
+        assert intersection[3] == r.bounds[3]
+
+        # With match_ref = True, right and top should match ref, left and bottom should match shifted +/- res
+        intersection = r.intersection(r_shifted, match_ref=True)
+        assert abs(intersection[0] - r_shifted.bounds[0]) < r.res[0]
+        assert abs(intersection[1] - r_shifted.bounds[1]) < r.res[1]
+        assert intersection[2] == r.bounds[2]
+        assert intersection[3] == r.bounds[3]
+
+        # -- Test with a non overlapping raster -- #
+        warnings.simplefilter("error")
+        transform_shifted = (
+            r.transform.a,
+            r.transform.b,
+            r.transform.c + (r.width + 1) * r.res[0],
+            r.transform.d,
+            r.transform.e,
+            r.transform.f,
+        )
+        r_nonoverlap = gu.Raster.from_array(r.data, crs=r.crs, transform=transform_shifted, nodata=r.nodata)
 
-        assert inters == (left, bottom, right, top)
+        with pytest.warns(UserWarning, match="Intersection is void"):
+            intersection = r.intersection(r_nonoverlap)
+            assert intersection == (0.0, 0.0, 0.0, 0.0)
```

Comment:
Great series of tests! :slightly_smiling_face:

Reply:
😄